server/src/main/java/io/crate/metadata/SimpleReference.java

```diff
@@ -160,6 +160,27 @@ public Reference getRelocated(ReferenceIdent newIdent) {
         );
     }
 
+    @Override
+    public Reference applyColumnOid(Metadata.ColumnOidSupplier oidSupplier) {
+        var oid = this.oid;
+        if (oid == COLUMN_OID_UNASSIGNED) {
+            oid = oidSupplier.nextOid();
+        }
+        return new SimpleReference(
+                ident,
+                granularity,
+                type,
+                columnPolicy,
+                indexType,
+                nullable,
+                hasDocValues,
+                position,
```

Comment:
Can we skip instance creation and return existing `Reference` if oid is already assigned? Same for geo/index/gen refs.

Reply:
Good point. This should never happen, but will ensure it.